examples/bank/bank.go

```diff
@@ -0,0 +1,214 @@
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	bank.approxNumTransactions++
+	return true
+}
+
+// Read the balances in all the accounts and return them.
+func (bank *Bank) readAllAccounts() []int64 {
+	balances := make([]int64, bank.numAccounts)
+	// Key Value Client initialization.
+	sender, err := client.NewHTTPSender(bank.servingAddr, testutils.NewTestBaseContext())
+	if err != nil {
+		log.Fatal(err)
+	}
+	kvClient := client.NewKV(nil, sender)
+	kvClient.User = storage.UserRoot
+	txnOpts := &client.TransactionOptions{Name: "Reading all balances"}
+	err = kvClient.RunTransaction(txnOpts, func(txn *client.Txn) error {
+		for i := 0; i < bank.numAccounts; i++ {
+			call := client.Get(proto.Key(makeAccountID(i)))
+			gr := call.Reply.(*proto.GetResponse)
+			if err := txn.Run(call); err != nil {
+				return err
+			}
+
+			if gr.Value != nil && gr.Value.Bytes != nil {
+				balances[i], err = strconv.ParseInt(string(gr.Value.Bytes), 10, 64)
+				if err != nil {
+					return err
+				}
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+	return balances
+}
+
+// continuouslyTransferMoney() keeps moving random amounts between
+// random accounts.
+func (bank *Bank) continuousMoneyTransfer() {
+	// Key Value Client initialization.
+	sender, err := client.NewHTTPSender(bank.servingAddr, testutils.NewTestBaseContext())
+	if err != nil {
+		log.Fatal(err)
+	}
+	kvClient := client.NewKV(nil, sender)
+	kvClient.User = storage.UserRoot
+
+	for {
+		from := makeAccountID(rand.Intn(bank.numAccounts))
+		to := makeAccountID(rand.Intn(bank.numAccounts))
+		exchange := rand.Int63n(100)
+		bank.moveMoney(kvClient, from, to, exchange)
+		// time.Sleep(time.Duration(rand.Int63n(10)) * time.Millisecond)
+	}
+}
+
+// Initialize all the bank balances
+func initBankAccounts(servingAddr string, numAccounts int, cash int64) {
+	// Key Value Client initialization.
+	sender, err := client.NewHTTPSender(servingAddr, testutils.NewTestBaseContext())
+	if err != nil {
+		log.Fatal(err)
+	}
+	kvClient := client.NewKV(nil, sender)
+	kvClient.User = storage.UserRoot
+
+	putOpts := client.TransactionOptions{Name: "Initialize bank"}
+	for i := 0; i < numAccounts; i++ {
+		err = kvClient.RunTransaction(&putOpts, func(txn *client.Txn) error {
+			key := makeAccountID(i)
+			txn.Prepare(client.Put(proto.Key(key), []byte(fmt.Sprintf("%d", cash))))
+			return nil
+		})
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+}
+
+func main() {
+	fmt.Printf("Keep moving money in between accounts\n\n")
+	security.SetReadFileFn(securitytest.Asset)
+	serv := server.StartTestServer(nil)
+	defer serv.Stop()
+	var bank Bank
+	bank.servingAddr = serv.ServingAddr()
+	bank.numAccounts = 100
+	const initCash int64 = 1000
+
+	// Initialize all the bank accounts
+	initBankAccounts(bank.servingAddr, bank.numAccounts, initCash)
+
+	// Start all the money transfer routines
+	const numTransferRoutines int = 10
+	for i := 0; i < numTransferRoutines; i++ {
+		go bank.continuousMoneyTransfer()
```

Comment:
For debugging, I'd start by removing this `go` so that the money transfers happen synchronously. If the app still fails, then something basic is going wrong.